react-components/src/architecture/base/commands/SettingsItem.ts

```diff
@@ -0,0 +1,24 @@
+/*!
+ * Copyright 2024 Cognite AS
+ */
+import { type BaseCommand } from './BaseCommand';
+
+export type SettingsItem = {
+  key: () => number;
+};
+
+export class CommandSettingsItem implements SettingsItem {
```

Comment:
I think this is not necessary and overcomplicated. All commands has their own unique id as a getter. You simple make a DividerCommand from BaseCommand, and override a minimum. Should be about nothing. Then you have everything you need, including the unique number